packages/editor/src/core/components/hooks/options.tsx

```diff
@@ -60,3 +60,27 @@ export const useConfiguredCellPlugin = (pluginId: string) => {
 export const useLang = () => {
   return useSelector(getLang);
 };
+
+/**
+ * @returns cell spacing for the current cell sub-tree
+ */
+export const useCellSpacing: () => [number, number] = () => {
+  return normalizeCellSpacing(useOptions().cellSpacing);
+};
+
+/**
+ * @returns a Provider component that can be used to override cell spacing for a subtree of cells
+ */
+export const useCellSpacingProvider = (
+  cellSpacingX: number,
+  cellSpacingY: number
+) => {
+  const options = useOptions();
+  return (({ children }) => (
+    <OptionsContext.Provider
+      value={{ ...options, cellSpacing: [cellSpacingX, cellSpacingY] }}
+    >
+      {children}
+    </OptionsContext.Provider>
+  )) as React.FC<unknown>;
+};
```

Comment:
that looks a bit complicted.

Should cellspacing override the spacing options for the whole subtree?

if not, you could change `useCellSpacing` to take a nodeId and it would read the plugin definition of that nodeId to check whether it defines custom cellSpacing

I am also not a fan of returning a Component in a hook. You could directly use `OptionContext.Provider`, thats simpler. The above hook has the additional problem of returning a new Component on every render, which will break memoization of components and lead to rerenders

Reply:
@macrozone
Yes, the idea is that a layout plugin should be able to override cell spacing in the entire subtree of its children. It's either a provider or passing settings down in props like it's done in HTMLRenderer.

The idea behind not using OptionContext.Provider in Cell/Inner explicitly was to hide that implementation detail inside the hooks file where that Context is declared. For the cell that uses it it's just "a provider" that does the trick -- it doesn't have to concern itself which context exactly is responsible and how to construct it.

I agree with the rerender observation. The component should memoized.
Let me know if you want me to create it explicitly inside Cell/Inner or just fix the hook.

Reply:
@macrozone , to be completely clear, my idea of a better component returning hook is something like that:

```
const [Provider, value] = useCellSpacingProvider(x, y);
...
return (
  ...
   <Provider value={value}>
      {whatever}
   </Provider>
  ...
)
```

`Provider` is an opaque provider (the hook will be returning `OptionContext.Provider`) while `value` is a memoized opaque value. Cell/Inner is just using those without concerning itself with the particulars of the implementation.

Reply:
i think memoizing the component is fine and if it works flaweless, we can do it like that.

i just have the feeling that this is not that idiomatic, but thats probably because i have rarely seen or thought about a hook returning a Component (hooks returning elements on the other hand are quite common). 

i agree with your reasoning though that its an implementation detail that can be hidden in the hook. So its resolved for me